x-pack/plugins/maps/public/classes/sources/vector_source/vector_source.tsx

```diff
@@ -145,9 +146,27 @@ export interface IVectorSource extends ISource {
    * Provide unique ids for managing source requests in Inspector
    */
   getInspectorRequestIds(): string[];
+
+  /**
+   * Syncs source specific styling with mbMap this allows custom sources to further style the map layers/filters
+   */
+  syncSourceStyle(mbMap:MbMap,layers:string[]):void
```

Comment:
Then this would be

`syncSourceStyle?(mbMap:MbMap, getLayers: () => string[]) : void`